test/server/server_test.cc

```diff
@@ -382,14 +385,22 @@ TEST_P(ServerInstanceImplTest, NoHttpTracing) {
   options_.service_cluster_name_ = "some_cluster_name";
   options_.service_node_name_ = "some_node_name";
   EXPECT_NO_THROW(initialize("test/server/empty_bootstrap.yaml"));
-  EXPECT_NE(nullptr, dynamic_cast<Tracing::HttpNullTracer*>(&server_->httpContext().tracer()));
+  EXPECT_NE(nullptr, dynamic_cast<Tracing::HttpNullTracer*>(tracer()));
+  EXPECT_EQ(nullptr, dynamic_cast<Tracing::HttpTracerImpl*>(tracer()));
 }
 
 TEST_P(ServerInstanceImplTest, ZipkinHttpTracingEnabled) {
   options_.service_cluster_name_ = "some_cluster_name";
   options_.service_node_name_ = "some_node_name";
   EXPECT_NO_THROW(initialize("test/server/zipkin_tracing.yaml"));
-  EXPECT_EQ(nullptr, dynamic_cast<Tracing::HttpNullTracer*>(&server_->httpContext().tracer()));
+  EXPECT_EQ(nullptr, dynamic_cast<Tracing::HttpNullTracer*>(tracer()));
+
+  // Note: there is no ZipkingTracerImpl object;
```

Comment:
You could probably expose the underlying driver somehow and dynamic cast to that, but not sure it's worth it.

Reply:
I thought of that and also felt like that was overkill. Really the way to go deeper on this is to have a zipkin integration test that shows zipkin syntax coming out of the server. But for the resolution of https://github.com/envoyproxy/envoy/issues/5241 I really wanted something that just looked at the tracer field.